MMOCoreORB/src/server/zone/managers/resource/resourcespawner/ResourceSpawner.cpp

```diff
@@ -340,20 +361,41 @@ bool ResourceSpawner::writeAllSpawnsToScript() {
 
 			writer->writeLine("		},");
 
+			writer->writeLine("		inSpawn = \"" + String::valueOf(inPhase) + "\",");
+			writer->writeLine("		deSpawnTime = \"" + String::valueOf(spawn->getDespawned()) + "\",");
+
 			writer->writeLine("		zoneRestriction = \"" + spawn->getZoneRestriction() + "\",");
 			writer->writeLine("		surveyToolType = " + String::valueOf(spawn->getSurveyToolType()) + ",");
 			writer->writeLine("		containerCRC = " + String::valueOf(spawn->getContainerCRC()) + ",");
 
 			writer->writeLine("	},");
 			writer->writeLine("");
+
+			if(String::valueOf(inPhase) == "1") {
+				swgcraft->write(spawn->getName()+ ",");
+				swgcraft->write(spawn->getClass(last));
```

Comment:
This variable name `last` is kind of confusing. Is it trying to dynamically determine the last non-empty spawn class index?

If so, can we rename it to be more clear? Perhaps `lastSpawnClassIndex`? Additionally, I don't think we need to declare it outside of the main for loop.

```cpp
int lastSpawnClassIndex = 0;

writer->writeLine("		classes = {");
for(int i = 0; i < 8; ++i) {
	String spawnClass = spawn->getClass(i);
	if(spawnClass != "") {
		lastSpawnClassIndex = i;
		String spawnClass2 = spawn->getStfClass(i);
		writer->writeLine("			{\"" + spawnClass + "\", \"" + spawnClass2 + "\"},");
	}
}
```

Since you only use the variable within the scope of the main for loop, you shouldn't need to define it on line 315.

Same goes for the `ghDump()` usage of the variable (if we keep these functions separate)